river/cluster/clustream.py

```diff
@@ -229,20 +243,27 @@ def learn_one(self, x, w=1.0):
         # Otherwise, closest micro-clusters are merged with each other.
         self._maintain_micro_clusters(x=x, w=w)
 
-        return self
+        # Apply incremental K-Means on micro-clusters after each time_gap
+        if self._timestamp % self.time_gap == self.time_gap - 1:
```

Comment:
I'm not sure I like the "timestamp" because it's not really a timestamp. I would use something like `_n_obs_seen`

Reply:
We inherited this name directly from the paper so that it makes more sense to the users. Since this variable will also be seen and taken into account by micro clusters, we prefer it to be kept that way in order to make more sense word-wise.

Reply:
Ok no problem then if it's the terminology used in the paper

Reply:
Thank you so much @MaxHalford!